qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -267,6 +267,9 @@ def run(
         # pylint: disable=arguments-differ
         if job_share_level:
             try:
```

Comment:
nit - we probably don't need to validate share level anymore since it'll be completely ignored.